File: Cool3DSidebarAnimation_OC/Cool3DSidebarAnimation_OC/HamburgerView.m
//
//  HamburgerView.m
//  Cool3DSidebarAnimation_OC
//
//  Created by Mazy on 2017/8/3.
//  Copyright © 2017年 Mazy. All rights reserved.
//

#import "HamburgerView.h"

@interface HamburgerView()

@property (nonatomic, strong) UIImageView *hamburgerView;

@end

@implementation HamburgerView

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        [self setupUI];
    }
    return self;
}

- (void)setupUI {
    self.hamburgerView = [[UIImageView alloc] initWithImage: [UIImage imageNamed: @"Hamburger"]];
    self.hamburgerView.userInteractionEnabled = YES;
    [self addSubview:self.hamburgerView];
}

- (void)rotate: (CGFloat)fraction {
    CGFloat angle = fraction * M_PI_2;
    self.hamburgerView.transform = CGAffineTransformMakeRotation(angle);
}

@end

File: Cool3DSidebarAnimation/Cool3DSidebarAnimation/HamburgerView.swift
//
//  HamburgerView.swift
//  Cool3DSidebarAnimation
//
//  Created by  Mazy on 2017/5/21.
//  Copyright © 2017年 Mazy. All rights reserved.
//

import UIKit

class HamburgerView: UIView {

    let imageView: UIImageView = UIImageView(image: UIImage(named: "Hamburger"))
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        config()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        config()
    }
    
    func config() {
        imageView.contentMode = .center
        addSubview(imageView)
    }
    
    func rotate(fraction: CGFloat) {
        let angle = Double(fraction) * M_PI_2
        imageView.transform = CGAffineTransform(rotationAngle: CGFloat(angle))
    }

}
